awx/ui_next/src/screens/Inventory/InventoryAdd/InventoryAdd.test.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { createMemoryHistory } from 'history';
+import { mountWithContexts, waitForElement } from '@testUtils/enzymeHelpers';
+
+import { InventoriesAPI, CredentialTypesAPI } from '@api';
+import InventoryAdd from './InventoryAdd';
+
+jest.mock('@api');
+
+CredentialTypesAPI.read.mockResolvedValue({
+  data: {
+    results: [
+      {
+        id: 14,
+        name: 'insights',
+      },
+    ],
+  },
+});
+
+describe('<InventoryAdd />', () => {
+  let wrapper;
+  let history;
+  beforeEach(async () => {
+    history = createMemoryHistory({ initialEntries: ['/inventories'] });
+    await act(async () => {
+      wrapper = mountWithContexts(<InventoryAdd />, {
+        context: { router: { history } },
+      });
+    });
+  });
+
+  afterEach(() => {
+    wrapper.unmount();
+  });
+
+  test('Initially renders successfully', () => {
+    expect(wrapper.length).toBe(1);
+  });
+  test('handleSubmit should call the api', async () => {
+    await waitForElement(wrapper, 'isLoading', el => el.length === 0);
+
+    wrapper.update();
+    await act(async () => {
+      wrapper.find('InventoryForm').prop('handleSubmit')({
+        name: 'Foo',
+        id: 1,
+        organization: 2,
+      });
```

Comment:
You could add some instance groups as well and ensure the associate calls are made correctly